### PR TITLE
FW-3229 The widget is initialising with empty header

### PR DESCRIPTION
**Pull request by Prochnost55**
### What do you want to achieve?
- added a check to prevent undefined from getting passed to JSON.parse().

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [x] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

- Please test the code with an website env where
1. the session storage is not available initially.
2. kommunicate is not present in local storage.

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 1]
```diff
@@ -7063,7 +7063,7 @@ var userOverride = {
                 isAgentOffline
             ) {
                 var userSession = KommunicateUtils.isSessionStorageAvailable()
-                    ? JSON.parse(sessionStorage.kommunicate)
+                    ? (sessionStorage.kommunicate ? JSON.parse(sessionStorage.kommunicate) : {})
```
Comment (gargshantnu) [inline webplugin/js/app/mck-sidebox-1.0.js:7066]: `KommunicateUtils.isSessionStorageAvailable()` vs `sessionStorage.kommunicate`
whats the difference?

Comment (Prochnost55) [reply webplugin/js/app/mck-sidebox-1.0.js:7066]: sessionStorage is available but it doesn't have `kommunicate` in storage
thus `kommunicate` becomes undefined and `JSON.parse(undefined)` gives error

Comment (gargshantnu) [reply webplugin/js/app/mck-sidebox-1.0.js:7066]: any ideas why its not present?

Comment (Prochnost55) [reply webplugin/js/app/mck-sidebox-1.0.js:7066]: The customers' website was using some tool to ask user permissions related to cookie and storage
so when the user visits for the 1st time there was no data in storage and also no permission to write data.

Comment (gargshantnu) [inline webplugin/js/app/mck-sidebox-1.0.js:7066]: can we move the new check to outer level? like combine both the checks as it seems similar

**Review by ChannuBS — APPROVED** [review]

**Review by gargshantnu — APPROVED** [review]

**Pull request merged**